tools/mtmd/mtmd-audio.cpp

```diff
@@ -0,0 +1,975 @@
+#include "mtmd-audio.h"
+
+#define MINIAUDIO_IMPLEMENTATION
+#define MA_NO_ENCODING
+#define MA_NO_DEVICE_IO
+#define MA_NO_RESOURCE_MANAGER
+#define MA_NO_NODE_GRAPH
+#define MA_NO_ENGINE
+#define MA_NO_GENERATION
+#define MA_API static
+#include "miniaudio.h"
```

Comment:
@ggerganov I initially use `dr_wav.h` here, but I struggled to write myself a resampling algorithm to downsample/upsample audio to 16KHz. I ended up using `miniaudio.h` here which provide decoding wav, mp3, flac, etc while also come with resampling built-in.

However, the caveat is that this single-header library is 3MB of code, and most of the components are disabled upon compilation as you see here.

What do you think about keeping this lib? I think the other components can be useful for TTS, as it allow us to play the generated audio without an external command.

Reply:
It's ok to use it - we also switched to `miniaudio.h` for the `whisper.cpp` examples: https://github.com/ggml-org/whisper.cpp/pull/2759.

Just make sure it will be used only in the examples/tools and that `libmtmd` will not depend on it to function.

Reply:
Currently `libmtmd.so` contains both `miniaudio.h` and `stb_image.h`, but they are considered as helpers and not part of the main API. The main API only takes PCM32 as input.

I think what I can do in a follow-up PR is to move the helper into its own header file and its own `.so`

Reply:
Yes, no problem. It's not important to do it immediately.